frontend/src/components/BCDataGrid/BCColumnSetFilter.jsx

```diff
@@ -0,0 +1,137 @@
+        instance.onFloatingFilterChanged(null, null)
+      })
+      return
+    }
+
+    props.parentFilterInstance((instance) => {
+      if (!props.multiple) {
+        const val = input ? input.name : ''
+        setCurrentValue(val)
+        instance.onFloatingFilterChanged('equals', val)
+      } else {
+        const filterArr = input.map((item) => item.name).join(', ')
+        setCurrentValue(filterArr)
+        instance.onFloatingFilterChanged('equals', filterArr)
+      }
+    })
+  }
+
+  useEffect(() => {
+    if (!optionsData) return
+    const optionsDataCopy = optionsData.map((option) => ({
+      name: option[props.apiOptionField ? props.apiOptionField : 'name']
+    }))
+    // if already loaded then disable re-load
+    if (options.length > 0) return
+    setOptions(optionsDataCopy)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [optionsData, props.apiOptionField])
+
+  return (
+    <Autocomplete
+      ref={inputRef}
+      multiple={props.multiple}
+      disableCloseOnSelect={props.disableCloseOnSelect}
+      onChange={onInputBoxChanged}
+      isOptionEqualToValue={(option, value) => option.name === value.name}
+      limitTags={1}
+      id="bc-column-set-filter"
+      className="ag-list ag-select-list ag-ltr ag-popup-child ag-popup-positioned-under"
+      role="list-box"
+      sx={{ width: columnWidth }}
+      options={options}
+      loading={optionsIsLoading}
+      autoHighlight
+      size="small"
+      getOptionLabel={(option) => option.name}
+      renderOption={(propsIn, option, { selected }) => (
+        <Box
+          component="li"
+          key={option}
+          className={
+            selected
+              ? 'ag-list-item ag-select-list-item selected'
+              : 'ag-list-item ag-select-list-item'
+          }
+          role="option"
+          sx={{ '& > img': { mr: 2, flexShrink: 0 } }}
+          {...propsIn}
+        >
+          {props.multiple && (
+            <Checkbox
+              color="primary"
+              className="ag-set-filter-item-checkbox ag-labeled ag-label-align-right ag-checkbox ag-input-field"
+              role="presentation"
+              sx={{ border: '2px solid primary' }}
+              icon={icon}
+              checkedIcon={checkedIcon}
+              style={{ marginRight: 8 }}
+              checked={selected}
+            />
+          )}
+          {option.name}
+        </Box>
+      )}
+      renderInput={(params) => (
+        <TextField
+          className="ag-text-field ag-input-field auto-select-filter"
+          role="presentation"
+          {...params}
+          label=""
+          value={currentValue}
+          variant="outlined"
+          size="small"
+          inputProps={{
+            ...params.inputProps,
+            autoComplete: 'new-password' // disable autocomplete and autofill
+          }}
+        />
+      )}
+    />
+  )
+})
+
+BCColumnSetFilter.displayName = 'BCColumnSetFilter'
```

Comment:
we should add propTypes to this component

Reply:
updated.